From 0973c900278424af008da0c428ca848c63f69481 Mon Sep 17 00:00:00 2001
From: Eneas U de Queiroz <cote2004-github@yahoo.com>
Date: Fri, 25 May 2018 16:27:02 -0300
Subject: [PATCH 11/11] Adds anonymous ciphers for openssl 1.1

This patch adds anonymous ciphers when compiling with openssl 1.1, so it
retains the same behaviour as previous versions of openssl.  That means,
it passes all tests, even though this is very insecure.

Signed-off-by: Eneas U de Queiroz <cote2004-github@yahoo.com>
---
 conserver/main.c  | 5 ++++-
 console/console.c | 5 ++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/conserver/main.c b/conserver/main.c
index 59cfeeb..71b59c4 100644
--- a/conserver/main.c
+++ b/conserver/main.c
@@ -116,6 +116,9 @@ int DH_set0_pqg(DH *dh, BIGNUM *p, BIGNUM *q, BIGNUM *g)
     return 1;
 }
 #define TLS_method SSLv23_method
+#define CIPHER_SEC0
+#else
+#define CIPHER_SEC0 ":@SECLEVEL=0"
 #endif /* OPENSSL_VERSION_NUMBER < 0x10100000L */
 
 SSL_CTX *ctx = (SSL_CTX *)0;
@@ -366,7 +369,7 @@ SetupSSL(void)
 	    }
 	    ciphers = "ALL:!LOW:!EXP:!MD5:!aNULL:@STRENGTH";
 	} else {
-	    ciphers = "ALL:!LOW:!EXP:!MD5:@STRENGTH";
+	    ciphers = "ALL:aNULL:!LOW:!EXP:!MD5:@STRENGTH" CIPHER_SEC0;
 	}
 	if (config->sslcacertificatefile != (char *)0) {
 	    STACK_OF(X509_NAME) * cert_names;
diff --git a/console/console.c b/console/console.c
index 3d9ddee..418f2ed 100644
--- a/console/console.c
+++ b/console/console.c
@@ -74,6 +74,9 @@ struct winsize ws;
 
 #if OPENSSL_VERSION_NUMBER < 0x10100000L
 #define TLS_method SSLv23_method
+#define CIPHER_SEC0
+#else
+#define CIPHER_SEC0 ":@SECLEVEL=0"
 #endif /* OPENSSL_VERSION_NUMBER < 0x10100000L */
 
 
@@ -131,7 +134,7 @@ SetupSSL(void)
 # if defined(REQ_SERVER_CERT)
 	    ciphers = "ALL:!LOW:!EXP:!MD5:!aNULL:@STRENGTH";
 # else
-	    ciphers = "ALL:!LOW:!EXP:!MD5:@STRENGTH"
+	    ciphers = "ALL:aNULL:!LOW:!EXP:!MD5:@STRENGTH" CIPHER_SEC0;
 # endif
 	}
 	SSL_CTX_set_verify(ctx, SSL_VERIFY_PEER, SSLVerifyCallback);
-- 
2.16.1

